index.php

```diff
@@ -1,26 +1,24 @@
 <?php
 require('vendor/autoload.php');
-define('AMQP_DEBUG', true);
-use PhpAmqpLib\Connection\AMQPConnection;
-use PhpAmqpLib\Message\AMQPMessage;
-$url = parse_url(getenv('CLOUDAMQP_URL'));
-$conn = new AMQPConnection($url['host'], 5672, $url['user'], $url['pass'], substr($url['path'], 1));
-$ch = $conn->channel();
 
-$exchange = 'amq.direct';
-$queue = 'basic_get_queue';
-$ch->queue_declare($queue, false, true, false, false);
-$ch->exchange_declare($exchange, 'direct', true, true, false);
-$ch->queue_bind($queue, $exchange);
+use Enqueue\AmqpBunny\AmqpConnectionFactory;
+use Interop\Amqp\AmqpMessage;
+use Interop\Amqp\AmqpQueue;
 
-$msg_body = 'the body';
-$msg = new AMQPMessage($msg_body, array('content_type' => 'text/plain', 'delivery_mode' => 2));
-$ch->basic_publish($msg, $exchange);
+$context = (new AmqpConnectionFactory(getenv('CLOUDAMQP_URL')))->createContext();
+$queue = $context->createQueue('basic_get_queue');
+$queue->addFlag(AmqpQueue::FLAG_DURABLE);
```

Comment:
how can you set a flag after the queue is declared? all queue arguments are immutable.

Reply:
On a broker's side, it is indeed immutable, though if we try to declare a queue with exactly the same parameters several times the broker won't complain. 

The `createQueue` method does not perform declaration. It creates a local queue object. There is a `declareQueue` method that does the job. 

The queue interop is insipred by Java JMS spec and follows it. Here's the doc for create queue method: https://docs.oracle.com/javaee/7/api/javax/jms/JMSContext.html#createQueue-java.lang.String-